apps/cli/src/commands/boot-challenger.ts

```diff
@@ -255,13 +272,25 @@ async function processMissedAssertions(commandInstance: Vorpal.CommandInstance)
             const assertionNode = await getAssertion(missedAssertionNodeNum);
             commandInstance.log(`[${new Date().toISOString()}] Missed assertion data retrieved. Starting the submission process...`);
 
+            // Check if enough time has passed that we can submit an assertion
+            const {isSubmitTime, currentChallenge} = await isChallengeSubmitTime();
+            
+            if(isSubmitTime) {
+
             await submitAssertionToReferee(
                 cachedSecretKey,
                 missedAssertionNodeNum,
                 assertionNode,
                 cachedSigner!.signer,
+                currentChallenge
             );
             commandInstance.log(`[${new Date().toISOString()}] Submitted assertion: ${missedAssertionNodeNum}`);
+            lastAssertionTime = Date.now();
+            return;
```

Comment:
We can't return without resetting `isProcessingMissedAssertions`

Reply:
Completely missed that, thank you for pointing it out.